### PR TITLE
Simplify ORC tests

### PR DESCRIPTION
**Pull request by dain**

### CODE REVIEW

**Comment by dain** [comment]
@electrum can you take a quick look at this?  It is just simplifies a bunch of orc test code.

File: presto-orc/src/test/java/com/facebook/presto/orc/TestingOrcPredicate.java [hunk 1]
```diff
@@ -200,12 +202,12 @@ protected boolean chunkMatchesStats(List<Double> chunk, ColumnStatistics columnS
             // statistics can be missing for any reason
             if (columnStatistics.getDoubleStatistics() != null) {
                 // verify min
-                if (!columnStatistics.getDoubleStatistics().getMin().equals(Ordering.natural().nullsLast().min(chunk))) {
+                if (Math.abs(columnStatistics.getDoubleStatistics().getMin() - Ordering.natural().nullsLast().min(chunk)) > 0.001) {
```
Comment (electrum) [inline presto-orc/src/test/java/com/facebook/presto/orc/TestingOrcPredicate.java]: Why would min/max be affected by rounding?

File: presto-orc/src/test/java/com/facebook/presto/orc/TestingOrcPredicate.java [hunk 2]
```diff
@@ -62,14 +64,14 @@ public static OrcPredicate createOrcPredicate(ObjectInspector objectInspector, I
             case SHORT:
             case INT:
             case LONG:
-                return new LongOrcPredicate(expectedValues);
+                return new LongOrcPredicate(expectedValues.stream().map(value -> value == null ? null : ((Number) value).longValue()).collect(toList()));
```
Comment (electrum) [inline presto-orc/src/test/java/com/facebook/presto/orc/TestingOrcPredicate.java]: Wrap all these

File: presto-orc/src/test/java/com/facebook/presto/orc/OrcTester.java [hunk 3]
```diff
@@ -439,6 +514,109 @@ static DataSize writeOrcColumn(File outputFile, Format format, RecordWriter reco
         return new DataSize(outputFile.length(), Unit.BYTE).convertToMostSuccinctDataSize();
     }
 
+    public static DataSize writeRcFileColumnOld(File outputFile, Format format, RecordWriter recordWriter, ObjectInspector columnObjectInspector, Iterator<?> values)
```
Comment (electrum) [inline presto-orc/src/test/java/com/facebook/presto/orc/OrcTester.java]: Typo? Should be `writeOrcFileColumnOld` not `Rc`?

File: presto-orc/src/test/java/com/facebook/presto/orc/OrcTester.java [hunk 4]
```diff
@@ -382,6 +385,78 @@ else if (skipFirstBatch && isFirst) {
...
+            Map<?, ?> actualMap = (Map<?, ?>) actual;
+            Map<?, ?> expectedMap = (Map<?, ?>) expected;
+            assertEquals(actualMap.size(), expectedMap.size());
+
+            Type keyType = type.getTypeParameters().get(0);
+            Type valueType = type.getTypeParameters().get(1);
+
+            List<Entry<?, ?>> expectedEntries = new ArrayList<>(expectedMap.entrySet());
+            for (Entry<?, ?> actualEntry : actualMap.entrySet()) {
+                for (Iterator<Entry<?, ?>> iterator = expectedEntries.iterator(); iterator.hasNext(); ) {
```
Comment (electrum) [inline presto-orc/src/test/java/com/facebook/presto/orc/OrcTester.java]: Make this a `while` loop

File: presto-orc/src/test/java/com/facebook/presto/orc/OrcTester.java [hunk 5]
```diff
@@ -382,6 +385,78 @@ else if (skipFirstBatch && isFirst) {
...
+                    try {
+                        assertColumnValueEquals(keyType, actualEntry.getKey(), expectedEntry.getKey());
+                        assertColumnValueEquals(valueType, actualEntry.getValue(), expectedEntry.getValue());
+                        iterator.remove();
+                    }
+                    catch (AssertionError ignored) {
+                    }
+                }
+            }
+            checkArgument(expectedEntries.isEmpty(), "Unmatched entries " + expectedEntries);
```
Comment (electrum) [inline presto-orc/src/test/java/com/facebook/presto/orc/OrcTester.java]: I think this should be `assertTrue`

File: presto-orc/src/test/java/com/facebook/presto/orc/TestingOrcPredicate.java (hunk 1 above)
Comment (dain) [inline presto-orc/src/test/java/com/facebook/presto/orc/TestingOrcPredicate.java]: float values have strange rounding

**Pull request closed without merging**